### PR TITLE
feat(IT-Wallet): [SIW-4173] How QR Code works banner and tracking

### PR DESCRIPTION
**Pull request by riccardosacco99**
## Short description
Adds an informational banner on the IT-Wallet QR code presentation screen ("Come funziona il QR Code?") with a link to the Help Center article and persistent dismiss.
Introduces Mixpanel tracking for the QR code page (`ITW_QR_CODE`) and for the banner lifecycle (`BANNER`, `TAP_BANNER`, `CLOSE_BANNER`) as per the analytics spec.

## List of changes proposed in this pull request
- UI
    - New `ItwProximityQrCodeInfoBanner` component shown below the QR code on `ItwProximityQrCodeScreen`.
    - Banner uses the help pictogram, neutral background, "Scopri di più" CTA that opens the Help Center article, and an x button to dismiss.
    - Dismiss is persistent forever, reusing the existing ITW banners slice with the new `proximity_qr_code_info` banner id.
    - New i18n keys under `features.itWallet.presentation.qrCode.banner` (IT, EN, DE).

- Analytics — banner events
    - `BANNER` (impression) emitted when the banner becomes visible.
    - `TAP_BANNER` emitted on "Scopri di più".
    - `CLOSE_BANNER` emitted on x.
    - All three carry `banner_id: "itwQrCodeInfos"`, `banner_page: "ITW_QR_CODE"`, `banner_landing: <help center URL>`.

- Analytics — QR code page event
    - `ITW_QR_CODE` (screen view) now emitted from the screen on every focus, once the page has reached a determined state.
    - Properties: source (`ITW_CREDENTIAL_DETAIL` or `WALLET_HOME`) — passed via new route param from the two entry points; `qr_code_status` (`valid` / `generation_failed` / `PID_expired`) — derived from screen state.
    - Trigger moved out of the XState machine action into the screen so that the `PID_expired` case is also covered (the machine never reaches the generation step in that path).

- Tests
    - New component test `ItwProximityQrCodeInfoBanner.test.tsx` (snapshot + impression tracking).
    - Extended `ItwProximityQrCodeScreen.test.tsx` with a `describe("ITW_QR_CODE tracking")` suite covering all 3 `qr_code_status` values + the "no source" and "still loading" edge cases.
    - Updated `machine/__tests__/actions.test.ts` to reflect the removal of the success-branch tracking in the XState action.
    - Updated existing screen snapshots to include the new banner.

## How to test
Prerequisites: active IT-Wallet with at least one presentable credential. Mixpanel debug log (or Flipper plugin) enabled to observe events.

- Banner
    1. From WalletHome, tap the "Mostra codice QR" CTA → the QR code screen opens with the info banner visible below the QR.
    2. Observe Mixpanel: `BANNER` event fires once on screen view with `banner_id: "itwQrCodeInfos"`, `banner_page: "ITW_QR_CODE"`, `banner_landing: https://assistenza.ioapp.it/....`
    3. Tap "Scopri di più" → the Help Center article opens in the browser; `TAP_BANNER` fires with the same properties.
    4. Go back to the app and tap the x on the banner → banner disappears; `CLOSE_BANNER` fires.
    5. Kill the app and reopen it → navigate to the QR code screen again → the banner does not reappear (persistent dismiss verified against AsyncStorage).

### CODE REVIEW

File: locales/de/index.json [hunk 1]
```diff
@@ -2483,6 +2483,14 @@
...
           "openPdf": "Dokument anzeigen",
           "shareButton": "Speichern oder freigeben"
+        },
+        "qrCode": {
+          "banner": {
+            "title": "Wie funktioniert QR Code?",
+            "content": "Es ermöglicht Ihnen, die digitalen Dokumente, die Sie dem Wallet hinzugefügt haben, verifizieren zu lassen.",
+            "action": "Mehr erfahren",
+            "invalid": "Bestätigen Sie Ihre Identität, um Ihre Dokumente weiterhin ohne Einschränkung zu verwenden."
+          }
```
Comment (mastro993) [inline locales/de/index.json]: We should not provide translations if they are not officially approved. This applies to both DE and EN translations. 

```suggestion
```

File: locales/en/index.json [hunk 2]
```diff
@@ -2201,6 +2201,14 @@
...
           "openPdf": "Show document",
           "shareButton": "Save or share"
+        },
+        "qrCode": {
+          "banner": {
+            "title": "How does QR Code works?",
+            "content": "It allows you to have the digital documents you have added to the Wallet verified.",
+            "action": "Learn more",
+            "invalid": "Confirm your identity to continue using your documents without limitation."
+          }
```
Comment (mastro993) [inline locales/en/index.json]: Ditto
```suggestion
```

File: ts/features/itwallet/presentation/proximity/analytics/enum.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ export enum ITW_PROXIMITY_SCREENVIEW_EVENTS {
   ITW_BLUETOOTH_BLOCK = "ITW_BLUETOOTH_BLOCK",
   ITW_BLUETOOTH_ACTIVATION = "ITW_BLUETOOTH_ACTIVATION",
   ITW_BLUETOOTH_NOT_ACTIVATED = "ITW_BLUETOOTH_NOT_ACTIVATED",
-  ITW_PROXIMITY_QR_CODE = "ITW_PROXIMITY_QR_CODE",
+  ITW_PROXIMITY_QR_CODE = "ITW_QR_CODE",
```
Comment (mastro993) [inline ts/features/itwallet/presentation/proximity/analytics/enum.ts]: Do we need this change?

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/itwallet/presentation/proximity/analytics/enum.ts (hunk 3 above)
Comment (riccardosacco99) [reply ts/features/itwallet/presentation/proximity/analytics/enum.ts]: I made this changhe to be consistent with the event `ITW_QR_CODE` expected from mixpanel https://www.figma.com/design/ADn7iuJTCGbBwBo31X51Tc/General_ITW-2026?node-id=1135-109346&t=MikK4SQCksj4F2yI-1

Comment (riccardosacco99) [reply ts/features/itwallet/presentation/proximity/analytics/enum.ts]: I'll revert it and add a new ITW_QR_CODE key

**Review by mastro993 — APPROVED** [review]
LGTM
